### PR TITLE
admin: impl admin check index for mv index

### PR DESCRIPTION
**Pull request by xiongjiwei**
Signed-off-by: xiongjiwei <xiongjiwei1996@outlook.com>

### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/40272

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: executor/builder.go [hunk 1]
```diff
@@ -433,7 +433,7 @@ func buildIndexLookUpChecker(b *executorBuilder, p *plannercore.PhysicalIndexLoo
 
 	tps := make([]*types.FieldType, 0, fullColLen)
 	for _, col := range is.Columns {
-		tps = append(tps, &(col.FieldType))
+		tps = append(tps, col.FieldType.ArrayType())
```
Comment (tangenta) [inline executor/builder.go:436]: Why is it `ArrayType()`?

Comment (xiongjiwei) [reply executor/builder.go:436]: the index data is encoded with the element type in the array, so we should read the index in the same type

Comment (tangenta) [reply executor/builder.go:436]: But for the other types, the clone is unnecessary and it is confusing.

Comment (xhebox) [reply executor/builder.go:436]: +1, this seems confusing for me. Need some comments maybe.

**Review by xhebox — APPROVED** [review]

File: executor/distsql.go [hunk 2]
```diff
@@ -1254,36 +1254,44 @@ func (w *tableWorker) compareData(ctx context.Context, task *lookupTableTask, ta
...
+						if err != nil {
+							return errors.Trace(err)
+						}
+						if cmpRes == 0 {
+							break
+						}
 					}
+				} else {
+					cmpRes, err = idxVal.Compare(sctx, &vals[i], collators[i])
+				}
```
Comment (tangenta) [inline executor/distsql.go:1276]: Could you extract this into a separate function?

Comment (xiongjiwei) [reply executor/distsql.go:1276]: ok

File: table/tables/index.go [hunk 3]
```diff
@@ -438,55 +438,57 @@ func GenTempIdxKeyByState(indexInfo *model.IndexInfo, indexKey kv.Key) (key, tem
...
-		tempKey []byte
-		keyVer  byte
-	)
-	// If index current is in creating status and using ingest mode, we need first
-	// check key exist status in temp index.
-	key, tempKey, keyVer = GenTempIdxKeyByState(c.idxInfo, key)
-	if keyVer != TempIndexKeyTypeNone {
-		KeyExistInfo, h1, err1 := KeyExistInTempIndex(context.TODO(), txn, tempKey, distinct, h, c.tblInfo.IsCommonHandle)
-		if err1 != nil {
+func (c *index) Exist(sc *stmtctx.StatementContext, txn kv.Transaction, indexedValue []types.Datum, h kv.Handle) (bool, kv.Handle, error) {
```
Comment (tangenta) [inline table/tables/index.go:441]:
```suggestion
func (c *index) Exist(sc *stmtctx.StatementContext, txn kv.Transaction, indexedValues []types.Datum, h kv.Handle) (bool, kv.Handle, error) {
```

Comment (xiongjiwei) [reply table/tables/index.go:441]: do not think it is a big problem

**Review by tangenta — APPROVED** [review]

**Pull request merged**